### ISSUE TITLE
upgrade to 1.0.7.3

### ISSUE THREAD

**Issue by Aglidic** [MrMEEE/awx-build#35]
Hello i try to upgrade to last version but it doesn't work.

yum upgrade 
sudo -u awx /opt/awx/bin/awx-manage makemigrations --merge -v3
No conflicts detected to merge.
sudo -u awx /opt/awx/bin/awx-manage migrate -v3
Operations to perform:
  Apply all migrations: auth, conf, contenttypes, djcelery, main, oauth2_provider, sessions, sites, social_django, sso, taggit
Running pre-migrate handlers for application auth
Running pre-migrate handlers for application contenttypes
Running pre-migrate handlers for application sessions
Running pre-migrate handlers for application sites
Running pre-migrate handlers for application oauth2_provider
Running pre-migrate handlers for application rest_framework
Running pre-migrate handlers for application django_extensions
Running pre-migrate handlers for application djcelery
Running pre-migrate handlers for application polymorphic
Running pre-migrate handlers for application taggit
Running pre-migrate handlers for application social_django
Running pre-migrate handlers for application conf
Running pre-migrate handlers for application main
Running pre-migrate handlers for application api
Running pre-migrate handlers for application ui
Running pre-migrate handlers for application sso
Running pre-migrate handlers for application solo
Running migrations:
  Applying main.0043_v330_oauth2accesstoken_modified...Traceback (most recent call last):
  File "/opt/awx/bin/awx-manage", line 11, in <module>
    sys.exit(manage())
  File "/opt/awx/embedded/lib/python2.7/site-packages/awx/__init__.py", line 116, in manage
    execute_from_command_line(sys.argv)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/operations/fields.py", line 88, in database_forwards
    field,
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/base/schema.py", line 429, in add_field
    self.execute(sql, params)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/base/schema.py", line 120, in execute
    cursor.execute(sql, params)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: column "modified" contains null values

**Comment by Aglidic** [comment]
this error was before modifying the files:
/opt/awx/embedded/lib/python2.7/site-packages/awx/main/migrations/0043_v330_oauth2accesstoken_modified.py

After modifying it i now have this:
[root@gvaawxapp01t a_anglidic@GVA.ICRC.Priv]# sudo -u awx /opt/awx/bin/awx-manage migrate -v3
Operations to perform:
  Apply all migrations: auth, conf, contenttypes, djcelery, main, oauth2_provider, sessions, sites, social_django, sso, taggit
Running pre-migrate handlers for application auth
Running pre-migrate handlers for application contenttypes
Running pre-migrate handlers for application sessions
Running pre-migrate handlers for application sites
Running pre-migrate handlers for application oauth2_provider
Running pre-migrate handlers for application rest_framework
Running pre-migrate handlers for application django_extensions
Running pre-migrate handlers for application djcelery
Running pre-migrate handlers for application polymorphic
Running pre-migrate handlers for application taggit
Running pre-migrate handlers for application social_django
Running pre-migrate handlers for application conf
Running pre-migrate handlers for application main
Running pre-migrate handlers for application api
Running pre-migrate handlers for application ui
Running pre-migrate handlers for application sso
Running pre-migrate handlers for application solo
Running migrations:
  Applying main.0048_auto_20180820_0729...Traceback (most recent call last):
  File "/opt/awx/bin/awx-manage", line 11, in <module>
    sys.exit(manage())
  File "/opt/awx/embedded/lib/python2.7/site-packages/awx/__init__.py", line 116, in manage
    execute_from_command_line(sys.argv)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/migrations/operations/fields.py", line 221, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/base/schema.py", line 515, in alter_field
    old_db_params, new_db_params, strict)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/postgresql/schema.py", line 112, in _alter_field
    new_db_params, strict,
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/base/schema.py", line 690, in _alter_field
    params,
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/base/schema.py", line 120, in execute
    cursor.execute(sql, params)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/opt/awx/embedded/lib64/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: column "modified" contains null values

**Comment by Aglidic** [comment]
my bad after reverting to snapshot and relaunch the process from migrations.md it works

**Comment by Aglidic** [comment]
Aug 20 10:15:01 gvaawxapp01t systemd[1]: Started Flexible Branding Service.
Aug 20 10:15:01 gvaawxapp01t polkitd[814]: Unregistered Authentication Agent for unix-process:2294:97119 (system bus name :1.32, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-
Aug 20 10:15:01 gvaawxapp01t systemd[1]: Starting Flexible Branding Service...
Aug 20 10:15:01 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2320]: root:[2216] systemctl daemon-reload [0]#
Aug 20 10:15:03 gvaawxapp01t polkitd[814]: Registered Authentication Agent for unix-process:2324:97315 (system bus name :1.33 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/Poli
Aug 20 10:15:03 gvaawxapp01t systemd[1]: Started AWX Celery worker service.
Aug 20 10:15:03 gvaawxapp01t systemd[1]: Starting AWX Celery worker service...
Aug 20 10:15:03 gvaawxapp01t polkitd[814]: Unregistered Authentication Agent for unix-process:2324:97315 (system bus name :1.33, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-
Aug 20 10:15:03 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2336]: root:[2216] systemctl restart awx-celery-worker [0]#
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,275 INFO     awx.main.tasks Syncing Schedules
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,347 DEBUG    awx.main.tasks Cluster node heartbeat task.
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,354 WARNING  awx.main.tasks Rejoining the cluster as instance gvaawxapp01t.
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,360 ERROR    awx.main.tasks Host gvaawxapp03t last checked in at 2018-08-20 08:10:18.332549+00:00, marked as lost.
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,373 WARNING  awx.main.tasks celery worker dynamic --autoscale=50,4
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,385 ERROR    awx.main.tasks celery -n celery@%(ENV_HOSTNAME)s does not match settings.CLUSTER_HOST_ID=gvaawxapp01t
Aug 20 10:15:05 gvaawxapp01t celery[2330]: [2018-08-20 08:15:05,387: ERROR/MainProcess] Unrecoverable error: RuntimeError(u'celery -n celery@%(ENV_HOSTNAME)s does not match settings.CLUSTER_HOST_ID=gvaawxapp
Aug 20 10:15:05 gvaawxapp01t celery[2330]: Traceback (most recent call last):
Aug 20 10:15:05 gvaawxapp01t celery[2330]: File "/opt/awx/embedded/lib/python2.7/site-packages/celery/worker/__init__.py", line 206, in start
Aug 20 10:15:05 gvaawxapp01t celery[2330]: self.blueprint.start(self)
Aug 20 10:15:05 gvaawxapp01t celery[2330]: File "/opt/awx/embedded/lib/python2.7/site-packages/celery/bootsteps.py", line 119, in start
Aug 20 10:15:05 gvaawxapp01t celery[2330]: self.on_start()
Aug 20 10:15:05 gvaawxapp01t celery[2330]: File "/opt/awx/embedded/lib/python2.7/site-packages/celery/apps/worker.py", line 157, in on_start
Aug 20 10:15:05 gvaawxapp01t celery[2330]: sender=self.hostname, instance=self, conf=self.app.conf,
Aug 20 10:15:05 gvaawxapp01t celery[2330]: File "/opt/awx/embedded/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
Aug 20 10:15:05 gvaawxapp01t celery[2330]: response = receiver(signal=self, sender=sender, **named)
Aug 20 10:15:05 gvaawxapp01t celery[2330]: File "/opt/awx/embedded/lib/python2.7/site-packages/awx/main/tasks.py", line 258, in auto_register_ha_instance
Aug 20 10:15:05 gvaawxapp01t celery[2330]: raise RuntimeError(error)
Aug 20 10:15:05 gvaawxapp01t celery[2330]: RuntimeError: celery -n celery@%(ENV_HOSTNAME)s does not match settings.CLUSTER_HOST_ID=gvaawxapp01t
Aug 20 10:15:05 gvaawxapp01t celery[2330]: 2018-08-20 08:15:05,393 WARNING  awx.main.tasks Normal shutdown signal for instance gvaawxapp01t, removed self from capacity pool.
Aug 20 10:15:08 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2361]: root:[2216] systemctl status awx-celery-worker [3]#
Aug 20 10:16:03 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2380]: root:[2216] systemctl status awx-celery-worker -l [3]#
Aug 20 10:16:12 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2389]: root:[2216] journalctl -f [130]#
Aug 20 10:16:15 gvaawxapp01t a_anglidic@GVA.ICRC.Priv[2396]: root:[2216] clear [0]#

**Comment by MrMEEE** [comment]
Hi.. 

I had that issue as well, thought it was something with my installation..

Workaround:

Change %(ENV_HOSTNAME)s  to your hostname in the Service file..

I'll be looking into this

**Comment by Aglidic** [comment]
ok i finally made it work. 
I don't know why but it's like the celery can't understand the %(ENV_HOSTNAME)s anymore.
i had to write -n gvaawxapp01t to make it work

**Comment by Aglidic** [comment]
thanks for the tips.
It seems we arrived at the same conclusion

**Comment by Aglidic** [comment]
i saw another error but on daphne.

it tolds me that some python modules are missing (http2 and tls)

**Comment by Aglidic** [comment]
ok so i rollback for the moment. I don't know if it's coming from my HA conf or not. But no job was running anymore.

have to debug that

**Comment by Aglidic** [comment]
here is the last error:
ERROR    awx.main.consumers access denied to channel, XRF mismatch for admin

**Comment by Aglidic** [comment]
hum it seems that it's due to the cluster config. here i found something on the db:
ERROR:  column main_instance.managed_by_policy does not exist at character 297
STATEMENT:  SELECT "main_instance"."id", "main_instance"."uuid", "main_instance"."hostname", "main_instance"."created", "main_instance"."modified", "main_instance"."last_isolated_check", "main_instance"."version", "main_instance"."capacity", "main_instance"."capacity_adjustment", "main_instance"."enabled", "main_instance"."managed_by_policy", "main_instance"."cpu", "main_instance"."memory", "main_instance"."cpu_capacity", "main_instance"."mem_capacity" FROM "main_instance" WHERE "main_instance"."hostname" = 'gvaawxapp01t' LIMIT 1

**Comment by MrMEEE** [comment]
Guess that's also part of the stuff that can't be upgraded.. and needs a workaround.. just like:
https://github.com/ansible/awx/issues/2173

**Comment by Aglidic** [comment]
hum ok so if i understand well. We can't upgrade between "major" version. The only way is to deploy a new stack and make an export/import (which means loosing data)...

This awx project is really weird for an upstream project... 

it would be nice to have other case for this upgrade path

**Comment by MrMEEE** [comment]
Well.. that is basically what the upstream awx are saying.. and also that they don't wanna support a RPM build..

**Comment by MrMEEE** [comment]
It seems like the column is there on my installation... could you try:

su - postgres
psql
\c awx
select "main_instance"."managed_by_policy" from main_instance;


I get:
```
awx=# select "main_instance"."managed_by_policy" from main_instance;
 managed_by_policy 
-------------------
 t
(1 row) 
```

**Comment by MrMEEE** [comment]
awx/main/migrations/0045_v330_instance_managed_by_policy.py should create the column...

**Comment by Aglidic** [comment]
hum ok. just to be sure. 
you upgrade rpm
you do a awx-manage makemigrations --merge
you insert the null=True to the file 
and you do awx-manage migrate

that's it right?

**Comment by MrMEEE** [comment]
basically yes.. but actually the time did it, I did:

upgrade rpm
awx-manage makemigrations (failed)
awx-manage makemigrations --merge
awx-manage migrate (failed
insert the null=True to the file
awx-manage migrate

Should be the same...

**Comment by Aglidic** [comment]
ok i have rollback an rerun upgrade path and yes the main_instance is here

**Comment by Aglidic** [comment]
new error:
ERROR    awx.main.consumers Request user is not authenticated to use websocket.

**Comment by Aglidic** [comment]
so new test.  This time everything seems fine.
All services on all node are green
[root@gvaawxapp01t a_anglidic@GVA.ICRC.Priv]# systemctl status awx-*
● awx-celery-worker.service - AWX Celery worker service
   Loaded: loaded (/usr/lib/systemd/system/awx-celery-worker.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-08-21 13:51:45 CEST; 5min ago
 Main PID: 1107 (python2)
   CGroup: /system.slice/awx-celery-worker.service
           ├─1107 python2 /opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower,gvaawxapp01t -n celery@gvaawxapp01t
           ├─2044 python2 /opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower,gvaawxapp01t -n celery@gvaawxapp01t
           ├─2045 python2 /opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower,gvaawxapp01t -n celery@gvaawxapp01t
           ├─2046 python2 /opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower,gvaawxapp01t -n celery@gvaawxapp01t
           └─2047 python2 /opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower,gvaawxapp01t -n celery@gvaawxapp01t

Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,554: DEBUG/MainProcess] ^-- substep ok
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,554: DEBUG/MainProcess] | Consumer: Starting event loop
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,557: WARNING/MainProcess] celery@gvaawxapp01t ready.
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,557: DEBUG/MainProcess] | Worker: Hub.register Autoscaler...
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,557: DEBUG/MainProcess] | Worker: Hub.register Pool...
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,558: DEBUG/MainProcess] basic.qos: prefetch_count->16
Aug 21 13:51:55 gvaawxapp01t celery[1107]: [2018-08-21 11:51:55,874: DEBUG/MainProcess] celery@gvaawxapp03t joined the party
Aug 21 13:53:07 gvaawxapp01t celery[1107]: [2018-08-21 11:53:07,469: DEBUG/MainProcess] pidbox received method hello(revoked={}, from_node=u'celery@gvaawxapp02t') [reply_to:{u'routing_key': ...-8d9138ebfbdd]
Aug 21 13:53:07 gvaawxapp01t celery[1107]: [2018-08-21 11:53:07,470: INFO/MainProcess] sync with celery@gvaawxapp02t
Aug 21 13:53:08 gvaawxapp01t celery[1107]: [2018-08-21 11:53:08,629: DEBUG/MainProcess] celery@gvaawxapp02t joined the party

● awx-cbreceiver.service - AWX cbreceiver service
   Loaded: loaded (/usr/lib/systemd/system/awx-cbreceiver.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-08-21 13:51:45 CEST; 5min ago
 Main PID: 1102 (python2)
   CGroup: /system.slice/awx-cbreceiver.service
           ├─1102 python2 /opt/awx/bin/awx-manage run_callback_receiver -v2
           ├─2039 python2 /opt/awx/bin/awx-manage run_callback_receiver -v2
           ├─2040 python2 /opt/awx/bin/awx-manage run_callback_receiver -v2
           ├─2041 python2 /opt/awx/bin/awx-manage run_callback_receiver -v2
           └─2042 python2 /opt/awx/bin/awx-manage run_callback_receiver -v2

Aug 21 13:51:45 gvaawxapp01t systemd[1]: Started AWX cbreceiver service.
Aug 21 13:51:45 gvaawxapp01t systemd[1]: Starting AWX cbreceiver service...

● awx-web.service - AWX web service
   Loaded: loaded (/usr/lib/systemd/system/awx-web.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-08-21 13:51:45 CEST; 5min ago
 Main PID: 1099 (uwsgi)
   CGroup: /system.slice/awx-web.service
           ├─1099 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...
           ├─1201 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...
           ├─1202 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...
           ├─1203 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...
           ├─1204 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...
           └─1205 /opt/awx/bin/uwsgi --socket 127.0.0.1:8050 --module=awx.wsgi:application --logto=/var/log/awx/web.log --die-on-term --vacuum --uid awx --gid awx --processes=5 --harakiri=120 --no-orphans...

Aug 21 13:51:45 gvaawxapp01t systemd[1]: Started AWX web service.
Aug 21 13:51:45 gvaawxapp01t systemd[1]: Starting AWX web service...

● awx-channels-worker.service - AWX channels worker service
   Loaded: loaded (/usr/lib/systemd/system/awx-channels-worker.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-08-21 13:51:54 CEST; 5min ago
 Main PID: 2061 (python2)
   CGroup: /system.slice/awx-channels-worker.service
           └─2061 python2 /opt/awx/bin/awx-manage runworker --only-channels websocket.* -v2

Aug 21 13:55:37 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:55:37,400 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive
Aug 21 13:56:16 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:16,296 - DEBUG - worker - Got message on websocket.receive (reply daphne.response.eBMwDPQnTc!RUZFfNhtRR)
Aug 21 13:56:16 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:16,297 - DEBUG - runworker - websocket.receive
Aug 21 13:56:16 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:16,297 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive
Aug 21 13:56:23 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:23,654 - DEBUG - worker - Got message on websocket.receive (reply daphne.response.eBMwDPQnTc!RUZFfNhtRR)
Aug 21 13:56:23 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:23,655 - DEBUG - runworker - websocket.receive
Aug 21 13:56:23 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:23,655 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive
Aug 21 13:56:39 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:39,113 - DEBUG - worker - Got message on websocket.receive (reply daphne.response.eBMwDPQnTc!RUZFfNhtRR)
Aug 21 13:56:39 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:39,114 - DEBUG - runworker - websocket.receive
Aug 21 13:56:39 gvaawxapp01t awx-manage[2061]: 2018-08-21 11:56:39,114 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive

● awx-daphne.service - AWX daphne service
   Loaded: loaded (/usr/lib/systemd/system/awx-daphne.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-08-21 13:54:59 CEST; 1min 55s ago
 Main PID: 2271 (python2)
   CGroup: /system.slice/awx-daphne.service
           └─2271 python2 /opt/awx/bin/daphne -b 127.0.0.1 -p 8051 awx.asgi:channel_layer

Aug 21 13:54:59 gvaawxapp01t systemd[1]: Started AWX daphne service.
Aug 21 13:54:59 gvaawxapp01t systemd[1]: Starting AWX daphne service...
Aug 21 13:55:01 gvaawxapp01t daphne[2271]: 2018-08-21 11:55:01,304 INFO     Starting server at tcp:port=8051:interface=127.0.0.1, channel layer awx.asgi:channel_layer.
Aug 21 13:55:01 gvaawxapp01t daphne[2271]: 2018-08-21 11:55:01,305 INFO     HTTP/2 support not enabled (install the http2 and tls Twisted extras)
Aug 21 13:55:01 gvaawxapp01t daphne[2271]: 2018-08-21 11:55:01,305 INFO     Using busy-loop synchronous mode on channel layer
Aug 21 13:55:01 gvaawxapp01t daphne[2271]: 2018-08-21 11:55:01,305 INFO     Listening on endpoint tcp:port=8051:interface=127.0.0.1


nothing on journalct

but all new job stay in pending state 

and nothing appear on log

**Comment by MrMEEE** [comment]
systemctl status rabbitmq-server ?

**Comment by Aglidic** [comment]
was fine also.
and rabbitmqctl cluster_status too.

For what i saw the issue was somewhere at daphne level. Like daphne never answer properly 

i try a telnet and it was litstening.

**Comment by MrMEEE** [comment]
hmmm...

Mine looks the same, but works... have you tried increasing verbosity??

```
[root@akiva ~]# systemctl status awx-daphne                                                                                                                                                                           
● awx-daphne.service - AWX daphne service                                                                                                                                                                             
   Loaded: loaded (/usr/lib/systemd/system/awx-daphne.service; disabled; vendor preset: disabled)                                                                                                                     
   Active: active (running) since søn 2018-08-19 21:57:35 UTC; 1 day 14h ago                                                                                                                                          
 Main PID: 734 (python2)                                                                                                                                                                                              
   CGroup: /system.slice/awx-daphne.service                                                                                                                                                                           
           └─734 python2 /opt/awx/bin/daphne -b 127.0.0.1 -p 8051 awx.asgi:channel_layer

aug 19 21:57:35 akiva.outerrim.lan systemd[1]: Started AWX daphne service.
aug 19 21:57:35 akiva.outerrim.lan systemd[1]: Starting AWX daphne service...
aug 19 21:57:42 akiva.outerrim.lan daphne[734]: 2018-08-19 21:57:42,341 INFO     Starting server at tcp:port=8051:interface=127.0.0.1, channel layer awx.asgi:channel_layer.
aug 19 21:57:42 akiva.outerrim.lan daphne[734]: 2018-08-19 21:57:42,342 INFO     HTTP/2 support not enabled (install the http2 and tls Twisted extras)
aug 19 21:57:42 akiva.outerrim.lan daphne[734]: 2018-08-19 21:57:42,342 INFO     Using busy-loop synchronous mode on channel layer
aug 19 21:57:42 akiva.outerrim.lan daphne[734]: 2018-08-19 21:57:42,342 INFO     Listening on endpoint tcp:port=8051:interface=127.0.0.1
```

**Comment by Aglidic** [comment]
yes with verbosity = 2
if i launch a job
here is the result of journalctl -f in version 1.0.7:
Aug 21 14:17:37 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:37,473 DEBUG    WebSocket daphne.response.fFfkUJvMQZ!GsuGLNKtwc open and established
Aug 21 14:17:37 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:37,473 DEBUG    WebSocket daphne.response.fFfkUJvMQZ!GsuGLNKtwc accepted by application
Aug 21 14:17:37 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:37,475 DEBUG    Sent WebSocket packet to client for daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:17:38 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:38,476 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:17:38 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:38,483 - DEBUG - worker - Got message on websocket.receive (reply daphne.response.fFfkUJvMQZ!GsuGLNKtwc)
Aug 21 14:17:38 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:38,484 - DEBUG - runworker - websocket.receive
Aug 21 14:17:38 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:38,484 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive
Aug 21 14:17:41 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:41,630 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:17:45 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:45,836 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:17:53 gvaawxapp02t daphne[3535]: 2018-08-21 12:17:53,193 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:17:53 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:53,196 - DEBUG - worker - Got message on websocket.receive (reply daphne.response.fFfkUJvMQZ!GsuGLNKtwc)
Aug 21 14:17:53 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:53,196 - DEBUG - runworker - websocket.receive
Aug 21 14:17:53 gvaawxapp02t awx-manage[2732]: 2018-08-21 12:17:53,196 - DEBUG - worker - Dispatching message on websocket.receive to awx.main.consumers.ws_receive
Aug 21 14:18:06 gvaawxapp02t daphne[3535]: 2018-08-21 12:18:06,700 DEBUG    Sent WebSocket packet to client for daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:18:07 gvaawxapp02t daphne[3535]: 2018-08-21 12:18:07,703 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc
Aug 21 14:18:08 gvaawxapp02t daphne[3535]: 2018-08-21 12:18:08,753 DEBUG    WebSocket incoming frame on daphne.response.fFfkUJvMQZ!GsuGLNKtwc


and here is same result in working version 1.0.6.47:
Aug 21 14:44:23 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:23,368 - DEBUG - worker - Got message on websocket.disconnect (reply daphne.response.twRLxrvqYs!pTYSXXhnmi)
Aug 21 14:44:23 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:23,368 - DEBUG - runworker - websocket.disconnect
Aug 21 14:44:23 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:23,368 - DEBUG - worker - Dispatching message on websocket.disconnect to awx.main.consumers.ws_disconnect
Aug 21 14:44:24 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:24,421 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!UxDFSJOmxY to WebSocket daphne.response.twRLxrvqYs!zifWsmAAnW
Aug 21 14:44:24 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:24,468 DEBUG    WebSocket daphne.response.twRLxrvqYs!zifWsmAAnW open and established
Aug 21 14:44:24 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:24,468 DEBUG    WebSocket daphne.response.twRLxrvqYs!zifWsmAAnW accepted by application
Aug 21 14:44:25 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:25,472 DEBUG    WebSocket closed for daphne.response.twRLxrvqYs!zifWsmAAnW
Aug 21 14:44:25 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:25,476 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!ymQsvTyLuc to WebSocket daphne.response.twRLxrvqYs!dRLGMTSWAp
Aug 21 14:44:25 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:25,521 DEBUG    WebSocket daphne.response.twRLxrvqYs!dRLGMTSWAp open and established
Aug 21 14:44:25 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:25,522 DEBUG    WebSocket daphne.response.twRLxrvqYs!dRLGMTSWAp accepted by application
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,357: INFO/MainProcess] Received task: awx.main.tasks.update_inventory_computed_fields[867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8]
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,357: INFO/MainProcess] Scaling down -1 processes.
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,358: DEBUG/MainProcess] TaskPool: Apply <function _fast_trace_task at 0x7f675efcc488> (args:(u'awx.main.tasks.update_inventory_computed_fields', u'867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8', [2, True], {}, {u'utc': True, u'is_eager': False, u'chord': None, u'group': None, u'args': [2, True], u'retries': 0, u'delivery_info': {u'priority': None, u'redelivered': False, u'routing_key': u'awx_private_queue', u'exchange': u'awx_private_queue'}, u'expires': None, u'hostname': 'celery@%(ENV_HOSTNAME)s', u'task': u'awx.main.tasks.update_inventory_computed_fields', u'callbacks': None, u'correlation_id': u'867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8', u'errbacks': None, u'timelimit': [None, None], u'taskset': None, u'kwargs': {}, u'eta': None, u'reply_to': u'c5dad837-82c9-3baa-a1a9-a48e509b3125', u'id': u'867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8', u'headers': {}}) kwargs:{})
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,358: DEBUG/MainProcess] basic.qos: prefetch_count->88
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,363: DEBUG/MainProcess] Task accepted: awx.main.tasks.update_inventory_computed_fields[867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8] pid:1986
Aug 21 14:44:26 gvaawxapp02t celery[1098]: 2018-08-21 12:44:26,377 DEBUG    awx.main.models.inventory Going to update inventory computed fields
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,377: DEBUG/Worker-4] Going to update inventory computed fields
Aug 21 14:44:26 gvaawxapp02t celery[1098]: 2018-08-21 12:44:26,430 DEBUG    awx.main.models.inventory Finished updating inventory computed fields
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,430: DEBUG/Worker-4] Finished updating inventory computed fields
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,435: INFO/MainProcess] Task awx.main.tasks.update_inventory_computed_fields[867f0148-f1f0-4cb1-95ad-c2a5a36fd7b8] succeeded in 0.0740054769999s: None
Aug 21 14:44:26 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:26,513 DEBUG    Message on unknown channel u'daphne.response.twRLxrvqYs!DPDTzDMpWJ' - discarding
Aug 21 14:44:26 gvaawxapp02t celery[1098]: [2018-08-21 12:44:26,593: DEBUG/MainProcess] pidbox received method dump_active(safe=False) [reply_to:{u'routing_key': u'f590635a-bfbd-3898-bdbd-44bd45746ffc', u'exchange': u'reply.celery.pidbox'} ticket:79377752-bf71-4b77-bc8e-0e6e0b2dcd75]
Aug 21 14:44:27 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:27,515 DEBUG    WebSocket closed for daphne.response.twRLxrvqYs!dRLGMTSWAp
Aug 21 14:44:27 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:27,520 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!mgrLmxZURi to WebSocket daphne.response.twRLxrvqYs!OtJaLXOkKM
Aug 21 14:44:27 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:27,521 - DEBUG - worker - Got message on websocket.connect (reply daphne.response.twRLxrvqYs!OtJaLXOkKM)
Aug 21 14:44:27 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:27,521 - DEBUG - runworker - websocket.connect
Aug 21 14:44:27 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:27,522 - DEBUG - worker - Dispatching message on websocket.connect to awx.main.consumers.ws_connect
Aug 21 14:44:27 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:27,534 ERROR    awx.main.consumers Request user is not authenticated to use websocket.
Aug 21 14:44:27 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:27,566 DEBUG    WebSocket daphne.response.twRLxrvqYs!OtJaLXOkKM open and established
Aug 21 14:44:27 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:27,566 DEBUG    WebSocket daphne.response.twRLxrvqYs!OtJaLXOkKM accepted by application
Aug 21 14:44:27 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:27,769 DEBUG    Message on unknown channel u'daphne.response.twRLxrvqYs!DPDTzDMpWJ' - discarding
Aug 21 14:44:28 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:28,077 DEBUG    Message on unknown channel u'daphne.response.twRLxrvqYs!DPDTzDMpWJ' - discarding
Aug 21 14:44:28 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:28,077 DEBUG    WebSocket closed for daphne.response.twRLxrvqYs!OtJaLXOkKM
Aug 21 14:44:28 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:28,079 - DEBUG - worker - Got message on websocket.disconnect (reply daphne.response.twRLxrvqYs!OtJaLXOkKM)
Aug 21 14:44:28 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:28,080 - DEBUG - runworker - websocket.disconnect
Aug 21 14:44:28 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:28,080 - DEBUG - worker - Dispatching message on websocket.disconnect to awx.main.consumers.ws_disconnect
Aug 21 14:44:29 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:29,084 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!wKkNNckukP to WebSocket daphne.response.twRLxrvqYs!AwUjYxBBEU
Aug 21 14:44:29 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:29,130 DEBUG    WebSocket daphne.response.twRLxrvqYs!AwUjYxBBEU open and established
Aug 21 14:44:29 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:29,130 DEBUG    WebSocket daphne.response.twRLxrvqYs!AwUjYxBBEU accepted by application
Aug 21 14:44:30 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:30,133 DEBUG    WebSocket closed for daphne.response.twRLxrvqYs!AwUjYxBBEU
Aug 21 14:44:30 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:30,137 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!oogxmODMag to WebSocket daphne.response.twRLxrvqYs!QSmgDaujrt
Aug 21 14:44:30 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:30,183 DEBUG    WebSocket daphne.response.twRLxrvqYs!QSmgDaujrt open and established
Aug 21 14:44:30 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:30,183 DEBUG    WebSocket daphne.response.twRLxrvqYs!QSmgDaujrt accepted by application
Aug 21 14:44:31 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:31,187 DEBUG    WebSocket closed for daphne.response.twRLxrvqYs!QSmgDaujrt
Aug 21 14:44:31 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:31,193 DEBUG    Upgraded connection daphne.response.twRLxrvqYs!zZIdkKutpE to WebSocket daphne.response.twRLxrvqYs!EPsgpPtHqu
Aug 21 14:44:31 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:31,196 - DEBUG - worker - Got message on websocket.connect (reply daphne.response.twRLxrvqYs!EPsgpPtHqu)
Aug 21 14:44:31 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:31,196 - DEBUG - runworker - websocket.connect
Aug 21 14:44:31 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:31,196 - DEBUG - worker - Dispatching message on websocket.connect to awx.main.consumers.ws_connect
Aug 21 14:44:31 gvaawxapp02t awx-manage[1101]: 2018-08-21 12:44:31,208 ERROR    awx.main.consumers Request user is not authenticated to use websocket.
Aug 21 14:44:31 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:31,237 DEBUG    WebSocket daphne.response.twRLxrvqYs!EPsgpPtHqu open and established
Aug 21 14:44:31 gvaawxapp02t daphne[1106]: 2018-08-21 12:44:31,237 DEBUG    WebSocket daphne.response.twRLxrvqYs!EPsgpPtHqu accepted by application

**Comment by Aglidic** [comment]
so for me daphne doesn't send the message or celery never receive it.

i don't know which one is the issue. That can also ne celery with the weird change of (ENV_HOSTNAME)s

**Comment by MrMEEE** [comment]
I think I have fixed the celery hostname issue.. try using this:
```
ExecStart=/opt/awx/bin/celery worker -A awx -B -l info --autoscale=50,4 -Ofair -s /var/lib/awx/beat.db -n celery@%H
```

Probably won't fix the other issue

**Comment by Aglidic** [comment]
hum i was thinking i have a big difference with you.

I have disable celery beat maybe it's mandatory now.

I can't do test before 10 september but i will try again with celery-beat

**Comment by MrMEEE** [comment]
ok.. let me hear how it goes

**Comment by powertim** [comment]
Hi,

I've tested the upgrade with the HA setup (sorry @Aglidic i couldn't wait your return ^^) and I had the same strange behavior with all jobs in pending state.
Then, I changed the /etc/systemd/system/multi-user.target.wants/awx-celery-worker.service file to modify ExecStart with
`ExecStart=/opt/awx/bin/celery worker -A awx -B -l info --autoscale=50,4 -Ofair -s /var/lib/awx/beat.db -n celery@%H`

So now every queue is added to celery.
It resolved the pending state for all jobs (including SCM updates for example)..
But for the moment it distributes between 2 nodes only.
I just rebooted the 3 nodes to check another time (tomorrow normally).
But all runs of the playbook was strange, took only 15 seconds (normally about 10 minutes) and the display is abnormal so I may encounter another issue :

Vault password: 
2
3 PLAY [all] *********************************************************************
18:00:49
4
5 TASK [Gathering Facts] *********************************************************
18:00:49
6 ok: [gvaawxapp16t]
7
8 TASK [include_role] ************************************************************
18:00:58
9
10 TASK [icrc.test : debug] *******************************************************
18:00:58
11 ok: [gvaawxapp16t] => {
12 "msg": "System gvaawxapp16t"
13 }
14
15
TASK [Start icrc.test] *********************************************************
18:00:58
16 ok: [gvaawxapp16t] => {
17 "msg": "System gvaawxapp16t"
18 }
19
20 PLAY RECAP *********************************************************************
18:00:58
21 gvaawxapp16t               : ok=3    changed=0    unreachable=0    failed=0   
22

Hope I'll find a solution.

Best,

Tim.

**Comment by Aglidic** [comment]
we disable beat in version 1.0.6 because it's not clustered.

Maybe the best way this time is to deploy an awx in HA with container and doing reverse engineering on them to see the commands.

Or just study the container build

**Comment by powertim** [comment]
I suppose that new/additional queues need to be added to the - Q parameter of the awx-celery-worker to recover a functional HA setup.
But reverse engineering from K8s looks like a good idea.

**Comment by Aglidic** [comment]
if beat is mandatory now. We have to create a beat cluster based on consul lock

if that's the case will be a lot of fun...

**Comment by Aglidic** [comment]
i found this: celery worker -A awx -l DEBUG -B -Ofair --autoscale=100,4 --schedule=$(CELERY_SCHEDULE_FILE) -Q tower_scheduler,tower_broadcast_all,$(COMPOSE_HOST),$(AWX_GROUP_QUEUES) -n celery@$(COMPOSE_HOST)

https://github.com/ansible/awx/commit/8faf5887754addeb4940f73a6e02a380f7623541

if i compare to what we didi we are missing some stuff :+1: 
AWX_GROUP_QUEUS,tower_broadcast_all

but we will need more check on that.

At least we can say that our issue is somewhere with celery

**Comment by powertim** [comment]
OK so finally everything works as before the upgrade.
The issue about the playbook was due to a change that I was not aware of, so don't mind ^^
Here is the configuration that is working as expected:

`ExecStart=/opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -n celery@%H`

So simple :-)
Beat is still not necessary for the HA setup or we are missing something that we don't see...

**Comment by Aglidic** [comment]
does it works now?

If yes can you try this:
ExecStart=/opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower_broadcast_all,%H -n celery@%H

or this:
ExecStart=/opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,%H -n celery@%H

In that case the only thing is to replace ENV_HOSTNAME with just H

**Comment by powertim** [comment]
Yes as I said it works now :-)
Tested with 30 deployments in parallel, 100% success.

> ExecStart=/opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,tower_broadcast_all,%H -n celery@%H
> or this:
> ExecStart=/opt/awx/bin/celery worker -A awx -l debug --autoscale=50,4 -Ofair -Q tower_scheduler,%H -n celery@%H

No it doesn't work with those arguments.

But I don't care cause everything's fine now :-D

**So to conclude, update to 1.0.7.3 complete, no data loss, no need to export and import data, all steps of
https://github.com/MrMEEE/awx-build/blob/master/migrations.md followed with success.**

**Comment by tjyang** [comment]
@powertim 
Thanks to you guys' effort on this thread.

I followed https://github.com/MrMEEE/awx-build/blob/master/migrations.md and able to upgrade from 1.0.6.47. to [1.0.7.3](https://user-images.githubusercontent.com/378638/44623503-d6139080-a895-11e8-8507-8f4e08116549.png)
